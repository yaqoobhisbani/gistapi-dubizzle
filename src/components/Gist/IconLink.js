import React from "react";
import styled from "styled-components";
import Octicon from "react-octicon";
import { colors } from "../../theme/colors";

const IconLink = ({ icon, label, link, style }) => {
  return (
    <Wrapper style={style}>
      <Octicon name={icon} style={iconStyle} />
      <Label href={link}>{label}</Label>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  cursor: pointer;
`;

const Label = styled.a`
  font-size: 16px;
  color: ${(props) => props.theme.primary};
  text-decoration: none;
  cursor: pointer;
  margin-left: 4px;
`;

const iconStyle = {
  color: colors.primary,
  fontSize: 16,
  fontWeight: "500",
};

export default IconLink;
