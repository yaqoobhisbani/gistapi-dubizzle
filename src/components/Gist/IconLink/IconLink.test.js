import React from "react";
import { render, fireEvent } from "@testing-library/react";
import IconLink from "./IconLink";

describe("IconLink", () => {
  test("renders icon and label with link", () => {
    const label = "Files";
    const icon = "code";
    const link = "http://localhost/";
    const { getByText, getByTestId } = render(
      <IconLink label={label} icon={icon} link={link} />
    );
    const iconElement = getByTestId("icon");
    const labelElement = getByText(label);
    expect(iconElement).toBeInTheDocument();
    expect(labelElement).toBeInTheDocument();
    expect(labelElement.href).toBe(link);
  });

  test("clicking label triggers link", () => {
    const label = "Files";
    const icon = "code";
    const link = "http://localhost/";
    const { getByText } = render(
      <IconLink label={label} icon={icon} link={link} />
    );
    const labelElement = getByText(label);
    fireEvent.click(labelElement);
    expect(window.location.href).toBe(link);
  });

  test("matches snapshot", () => {
    const label = "Files";
    const icon = "code";
    const link = "http://localhost/";
    const { container } = render(
      <IconLink label={label} icon={icon} link={link} />
    );
    expect(container).toMatchSnapshot();
  });
});
