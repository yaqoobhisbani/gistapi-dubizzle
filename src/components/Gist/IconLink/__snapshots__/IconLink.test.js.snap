// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLink matches snapshot 1`] = `
<div>
  <div
    class="sc-beqWaB juKUET"
  >
    <span
      class="octicon octicon-code"
      data-testid="icon"
      style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
    />
    <a
      class="sc-gueYoa fnOQUP"
      href="http://localhost/"
    >
      Files
    </a>
  </div>
</div>
`;
